__main__.py

```diff
@@ -0,0 +1,211 @@
+import sys
+from cli import parse_args, validate_paths
+
+import pandas as pd
+import os
+
+import json
+from pathlib import Path
+
+def orginal_code(input_path, output_directory):
+    #!/usr/bin/env python
+    # coding: utf-8
+
+    # In[1]:
+
+    codeQL_directory = "/codeQL_data"
+    codeQL_directory = input_path+codeQL_directory
+    print(codeQL_directory)
+    clojure_directory = "/clojure_data/atomsClojure.csv"
+    clojure_directory = input_path+clojure_directory
+    coccinelle_directory = "/coccinelle_data"
+    coccinelle_directory = input_path+coccinelle_directory
+    dataframes = {}
+
+
+    # In[2]:
+
+
+    # Load CodeQL .sarif files and store them in dataframes dictionary
+
+    for filename in os.listdir(codeQL_directory):
+        if filename.endswith('.sarif'):
+            file_path = os.path.join(codeQL_directory, filename)
+            dataframe_key = filename[:-6]
+            pt = Path(file_path)
+            file_content = pt.read_text()
+            json_value = json.loads(file_content)
+            p = pd.DataFrame(columns = ["Type", "File", "Line", "CQL", "CQL_Column", "CQL_CodeQL"])
+            rows = []
+        print("Loading:", filename)
+        # These are the index paths of the data we are looking for. Each is located inside nested dictionaries.
+        for i in range(len(json_value["runs"][0]["results"])):
+            s = pd.Series({
+                "Type": filename[:-6],
+                "File": json_value["runs"][0]["results"][i]["locations"][0]["physicalLocation"]["artifactLocation"]["uri"], 
+                "Line": json_value["runs"][0]["results"][i]["locations"][0]["physicalLocation"]["region"]["startLine"],
+                "CQL": '+',
+                "CQL_Column": json_value["runs"][0]["results"][i]["locations"][0]["physicalLocation"]["region"].get("startColumn", None),
+                "CQL_Code": json_value["runs"][0]["results"][i]["locations"][0]["physicalLocation"]["contextRegion"]["snippet"]["text"].splitlines()[2:-2],
+            })
+            rows.append(s)
+        p = pd.concat(rows, axis=1).T
+        
+        # These lines convert these columns to string format
+        p["CQL_Code"] = p["CQL_Code"].apply(lambda x: '\n'.join(x) if isinstance(x, list) else x)
+        p["File"] = p["File"].apply(lambda x: '\n'.join(x) if isinstance(x, list) else x)
+        p["File"] = p["File"].str.split('/').str[-1]
+        dataframes[dataframe_key] = p
+        
+    print("DataFrame Names:", dataframes.keys())
+    # for i in dataframes.values():
+    #     display(i)
+
+
+    # In[3]:
+
+
+    # Load Clojure dataset
+
+    clojure_master = pd.read_csv(clojure_directory)
+    # Get only the filename, removing the path
+    clojure_master['file'] = clojure_master['file'].str.split('/').str[-1]
+    # display(clojure_master)
+    print("Clojure Types:", clojure_master["atom"].unique())
+
+
+    # In[4]:
+
+
+    # Prepare Clojure dataset for merge
+
+    name_mapping = {
```

Comment:
Sorry I am not too familiar with production practices in Python, please bear with my dumb questions here :)

I understand we want to move the dull set of name mappings to another place, is the purpose of this to make the code easier to read? 

If so, may I just copy and paste the mappings to a new file, named "mappings" for easier maintaince later on. However, soon as we standardize the namings in the input datasheet, no such mappings will no longer be necessary right?

Reply:
In my opinion, all constants and static mappings should be moved to the top of the module, so that they can be used in multiple functions without having to be specified several times and that it's easier to update them. I'd probably move them to the top of the module in this case:

```
NAME_MAPPING = {
...
```

Reply:
I got you, that will be an easy fix